openpype/hosts/houdini/api/pipeline.py

```diff
@@ -397,3 +398,24 @@ def _set_context_settings():
 
     lib.reset_framerange()
     lib.update_houdini_vars_context()
+
+
+def generate_thumbnail():
+    """
+    Generate Thumbnail from the current sceneview tab
+    """
+
+    if IS_HEADLESS:
+        log.debug("Generating thumbnail skipped due to headless mode")
+        return
+
+    host = registered_host()
+    if not host.get_current_workfile():
+        log.error("No current workfile path. Thumbnail generation skipped")
+        return
+
+    thumbnail_path = "{}_thumbnail.jpg".format(hou.text.expandString("$HIPNAME"))
+    sceneview = lib.get_scene_viewer()
+    lib.sceneview_snapshot(sceneview, thumbnail_path)
+
+    return thumbnail_path
```

Comment:
I'm not sure this should be part of this PR. I think this should remain in https://github.com/ynput/OpenPype/pull/5425 and potentially be updated there if this PR gets merged first - so it uses the same base functions from `lib`.